### DISCUSSION TITLE
when I add  tower::limit::RateLimitLayer in the route_layer, it shows the trait bound `tower::limit::RateLimit<Route<_>>: Clone` is not satisfied the trait `Clone` is not implemented for `tower::limit::RateLimit<Route<_>>`.

### DISCUSSION THREAD

**Discussion by goldwind-ting** [tokio-rs/axum#877] [Q&A] [answered]
### Version
axum = "0.4.8"
tower = { version = "0.4.12", features = ["timeout", "limit"] }
rustc 1.59.0

### Platform
linux ubuntu 21.10

### Description
when I use this method to add route_layer, it show follow error:
```the trait bound `tower::limit::RateLimit<Route<_>>: Clone` is not satisfied
the trait `Clone` is not implemented for `tower::limit::RateLimit<Route<_>>` ```.

```rust
use tower::limit::{ConcurrencyLimitLayer, RateLimitLayer};
Router::new()
        .route("/", get(health))
        .route("/v1/signup", post(signup).route_layer(RateLimitLayer::new(1000, std::time::Duration::from_secs(1))))
```
but, when I replace `RateLimitLayer` with `ConcurrencyLimitLayer`, it is normal.

**Comment by davidpdrsn** [comment, marked as answer]
Well the compile error has all the info. Services used with axum must be `Clone` which `RateLimit` isn't. I'm not familiar with how `RateLimit` so can't say why that is. You can combine it with [`BufferLayer`](https://docs.rs/tower/latest/tower/buffer/index.html) to make it `Clone`.
